lib/utils.js

```diff
@@ -141,16 +145,34 @@ exports.getXpiInfoFromManifest = getXpiInfoFromManifest;
 function getXpiInfo(xpiPath) {
   return extractXPI(xpiPath)
     .then(function(tmpXPI) {
-      var manifest = {};
-      try {
-        manifest = require(path.join(tmpXPI.path, "package.json"));
-      } catch (e) {}
-
-      tmpXPI.remove();
+      var getInfo;
+      var packageFile = path.join(tmpXPI.path, "package.json");
+      var installRdfFile = path.join(tmpXPI.path, "install.rdf");
 
-      // TODO: when there's no manifest, parse install.rdf to get info.
+      if (fileExists(packageFile)) {
```

Comment:
I feel like there could be cases where there is a package.json but it isn't a jpm.

Is there something in the file you could check to verify it is a jpm and have this fail if the file doesn't exist or it isn't a jpm?

I'm thinking something like:

```js
getInfo = getXpiInfoFromManifest(require(packageFile))
  .catch(ManifestError, function(err) {
      return getXpiInfoFromInstallRdf(data);
  }).catch(ManifestError, function(err) {
      throw new Error("Cannot get info: no manifest found in this XPI");
  });